frontend/catalyst/jax_tracer.py

```diff
@@ -847,15 +858,23 @@ def is_leaf(obj):
                 return_values, is_leaf=is_leaf
             )
 
-            # TODO: In order to compose transforms, we would need to recursively
-            # call apply_transform while popping the latest transform applied,
-            # until there are no more transforms to be applied.
-            # But first we should clean this up this method a bit more.
-            tapes, post_processing = apply_transform(qnode, quantum_tape, return_values_flat)
+            if isinstance(device, qml.devices.Device):
+                transform_program, _ = device.preprocess()
+            else:
+                transform_program = TransformProgram()
+
+            # We add pragma because lit test are giving qfunc directly
+            # But lit tests are not sending coverage results
+            if qnode:  # pragma: no branch
+                transform_program = qnode.transform_program + transform_program
+
+            tapes, post_processing = apply_transform(
+                transform_program, quantum_tape, return_values_flat
+            )
 
         # (2) - Quantum tracing
         transformed_results = []
-        is_program_transformed = qnode and qnode.transform_program
+        is_program_transformed = transform_program
```

Comment:
Not needed, but if you have the time to, what would you think of adding this bugfix? https://github.com/PennyLaneAI/catalyst/pull/574